middleware/auth.global.ts

```diff
@@ -0,0 +1,18 @@
+export default defineNuxtRouteMiddleware(async (to, from) => {
+  const app = useNuxtApp()
+  const {localStorage} = window;
+
+  if (!app.$appSettings.auth.enabled) {
+    return;
+  }
+
+  if (to.name !== 'login' && !app.$authToken.token) {
+    return await navigateTo('/login')
+  }
+
+  if (to.name === 'login' && to?.query?.token) {
+    localStorage?.setItem('token', to.query.token);
+    app.$authToken.token = to.query.token;
+    return await navigateTo('/')
+  }
```

Comment:
Question? Do we have a time limit for a token? If yes, we need to add Chaco for expiries date. Also we may need the logic to show authentic error and reconnection / reauth for cases where token not expired. And we need to clear storage if the token is expired